src/ethereum_test_forks/forks/forks.py

```diff
@@ -12,6 +12,13 @@ class Frontier(BaseFork):
     Frontier fork
     """
 
+    @classmethod
+    def fork(cls, block_number: int, timestamp: int) -> str:
```

Comment:
Point 1. How about specifying keyword arguments and providing defaults here, instead of defining them in `transition_tool.py`? I appreciate they belong more to the t8n tool, but as parameters required by the framework, having them here would keep the transition tool code simpler?
```suggestion
    def fork(cls, block_number: int = ALWAYS_TRANSITIONED_BLOCK_NUMBER, timestamp: int = ALWAYS_TRANSITIONED_BLOCK_TIMESTAMP) -> str:
```

Reply:
Awesome! Let me change this throughout the changes :)

Reply:
Changed the default values to `ALWAYS_TRANSITIONED_BLOCK_*` but only on transition forks, the default for the rest is zero. It worked out pretty nice because I also changed the way the transition decorator creates the methods.